recipes/powershell4.rb

```diff
@@ -30,14 +30,25 @@
     # Ensure .NET 4.5 is installed or installation will fail silently per Microsoft. Only necessary for Windows 2008R2 or 7.
     include_recipe 'ms_dotnet45' if windows_version.windows_server_2008_r2? || windows_version.windows_7?
 
-    # Reboot if user specifies immediate_reboot
-    include_recipe 'powershell::windows_reboot' if node['powershell']['installation_reboot_mode'] == 'immediate_reboot'
+    # Get powershell version
+    powershell_version_cmd = 'powershell.exe $PSVersionTable.PSVersion.Major'
+    shell_out = Mixlib::ShellOut.new(powershell_version_cmd)
+    shell_out.run_command
+    powershell_version = shell_out.stdout.to_f
+
+    # Reboot if user specifies doesn't specify no_reboot and powershell version < 4
+    include_recipe 'powershell::windows_reboot' unless node['powershell']['installation_reboot_mode'] == 'no_reboot' && powershell_version < 4
```

Comment:
Thanks!

I was actually thinking that we could take out the powershell version check as the "not_if" guard on the windows_package should ensure that we don't notify if the package hasn't been installed by chef. Happy to remove the extra check. For consistency at least.